rs/tests/networking/firewall_max_connections_test.rs

```diff
@@ -1,8 +1,179 @@
-#[rustfmt::skip]
+/* tag::catalog[]
+Title:: Firewall limit connection count.
+
+Goal:: Verify that nodes set a hard limit on number of simultaneous connections from a single IP addresses as defined in the firewall.
+
+Runbook::
+. Set up a test net, application typ, with 2 nodes.
+. Set up a universal vm with default config.
+. Set `max_simultaneous_connections_per_ip_address` to the configured value `max_simultaneous_connections_per_ip_address` in template file
+`ic.json5.template`.
+. Create `max_simultaneous_connections_per_ip_address` tcp connections from the driver simultaneously to a node and keep the connections alive.
+. Verify that the universal vm can create a tcp connection the node.
+. Verify the driver is unable to create new tcp connections to the node.
+. Terminate one of the active connections the driver has to the node.
+. Verify the node now accepts one connection at a time on ports [8080, 9090, 9091, 9100] from the driver.
+. All connectivity tests succeed as expected
+
+end::catalog[] */
 
 use anyhow::Result;
-use ic_system_test_driver::{driver::group::SystemTestGroup, systest};
-use ic_tests::networking::firewall_max_connections::{config, connection_count_test};
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::{
+    driver::{
+        group::SystemTestGroup,
+        ic::{InternetComputer, Subnet},
+        test_env::TestEnv,
+        test_env_api::{HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer, SshSession},
+        universal_vm::{UniversalVm, UniversalVms},
+    },
+    systest,
+    util::block_on,
+};
+use slog::{debug, info};
+use std::net::IpAddr;
+use std::time::Duration;
+use tokio::net::TcpStream;
+
+// This value reflects the value `max_simultaneous_connections_per_ip_address` in the firewall config file.
+const MAX_SIMULTANEOUS_CONNECTIONS_PER_IP_ADDRESS: usize = 1000;
+
+const UNIVERSAL_VM_NAME: &str = "httpbin";
+
+const TCP_HANDSHAKE_TIMEOUT: Duration = Duration::from_secs(2);
+
+fn config(env: TestEnv) {
```

Comment:
It would be great if you could rename this function to `setup(env: TestEnv)` because that's the modern name for these functions that set up the testnet.

Reply:
Done thanks

Reply:
Thanks!